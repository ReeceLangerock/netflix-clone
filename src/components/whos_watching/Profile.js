import PropTypes from 'prop-types';
import React from 'react'
import styled from 'styled-components'
import sg from './../../styleguide.js'

import bg from './../../assets/rick.jpg'
import meeseeks from './../../assets/meeseeks.jpg'
// import morty from './../../assets/morty.png'

const Avatar = styled.div`
height: 10vw;
width: 10vw;
max-height: 200px;
max-width: 200px;
min-height: 84px;
min-width: 84px;
border: .3em solid rgba(0,0,0,.4);
box-sizing: border-box;
position: relative;
text-decoration: none;
background-image: url(${props => (props.img ? require('./../../assets/' + props.img) : 'none')});
background-repeat: no-repeat;
background-size: contain;
background-color: ${props => (props.img ? '#333' : '')};
border: ${props => (props.img ? '.3em solid rgba(0,0,0,.4)' : 'none')};

display: flex;
justify-content: center;
align-items: center;
i{font-size: 6vw;
}

`

// ${props => 'url'+(require('./../../assets/' + props.userName.toLowerCase() + '.png'))};
const Container = styled.div`
margin: 0 2vw 0 0;
display: flex;
flex-direction: column;
align-content: center;
color: grey;
:last-child{
    margin: 0;
}

:hover {      
      ${Avatar}  {
        background-color: #e5e5e5;
        border-color: #e5e5e5;
        
        i {
          color: grey;
        }
        }
        
    }


`
const Name = styled.div`
line-height: 1.2em;
min-height: 2.4em;
display: block;
text-align: center;
font-size: 1.3vw;
margin: .8em 0;
text-overflow: ellipsis;
overflow: hidden;


`

export default class Profile extends React.Component {
  componentWillMount () {}
  render () {
    return (
      <Container>
        <Avatar img={this.props.img}>
          {!this.props.img && <i className={`fa fa-plus-circle`} aria-hidden='true' />}
        </Avatar>
        <Name>{this.props.userName}</Name>

      </Container>
    )
  }
}

Profile.propTypes = {
  userName: PropTypes.string.isRequired,
  img: PropTypes.string
};