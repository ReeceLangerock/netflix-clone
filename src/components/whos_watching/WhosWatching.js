import React from 'react'
import styled from 'styled-components'
import sg from './../../styleguide.js'
import { connect } from 'react-redux'

import * as actions from './../../redux/actions/actions'

import TopBar from './../general/TopBar'
import Profile from './Profile'

const Page = styled.section`
display: flex;
justify-content: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  content:'';
  z-index: 1;
  background: #141414;

  `

const Container = styled.div`
padding-top: 50px;
margin: 0 auto;
justify-content: center;
align-items: center;
display: flex;
flex-direction: column;
max-width: 80%;
text-align: center;
`

const ProfileContainer = styled.div`
display: flex;
margin: 2em 0;

`

const Title = styled.div`
width: 100%;
color: #fff;
font-size: 3.5vw;
`

const Button = styled.span`
display: block;
margin: 2em 0 1em 0;
font-size: 1.3vw;
color: grey;
border: 1px solid grey;
text-transform: uppercase;
padding: .5em 1.5em;
letter-spacing: 2px;
cursor: pointer;
font-size: .9em;


:hover {
  
  border-color: #e5e5e5;
  color: #e5e5e5;

  
}
`

export class WhosWatching extends React.Component {
  componentWillMount () {
   
    
  }
  render () {
    let loc = this.props.location.pathname.split('/')[1];
    return (
      <Page>
        <TopBar location = {loc}/>

        <Container>
          <Title>Who's Watching? </Title>

          <ProfileContainer>
            <Profile img='rick.jpg' userName='Rick' />
            <Profile img='morty.png' userName='Morty' />
            <Profile img='meeseeks.jpg' userName='Mr. Meeseeks' />
            <Profile userName='Add Profile' />

          </ProfileContainer>

          <Button>Manage Profiles</Button>
        </Container>

      </Page>
    )
  }
}
const mapStateToProps = state => ({})
export default connect(mapStateToProps, actions)(WhosWatching)
