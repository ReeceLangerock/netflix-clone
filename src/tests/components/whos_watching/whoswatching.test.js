import React from 'react'
import ReactDOM from 'react-dom'
import { WhosWatching } from './../../../components/whos_watching/WhosWatching'

describe('WhosWatching', () => {
  let props

  beforeEach(() => {
    props = {
      location: {
        pathname: 'test'
      }
    }
  })

  it('renders without crashing', () => {
    shallow(<WhosWatching {...props} />)
  })

  // it('should render a <i> if no image prop', () => {
  //   const wrapper = mount(<Profile userName='name' />)
  //   expect(wrapper.find('div').find('i').length).toBe(1)
  // })

  // it('should not render a <i> image is included', () => {
  //   const wrapper = mount(<Profile userName='name' img='meeseeks.jpg' />)
  //   expect(wrapper.find('div').find('i').length).toBe(0)
  // })

  it('matches snapshot', () => {
    const wrapper = shallow(<WhosWatching {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
