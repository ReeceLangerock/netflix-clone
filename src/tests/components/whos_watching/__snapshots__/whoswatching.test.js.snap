// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhosWatching matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WhosWatching
    location={
      Object {
        "pathname": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TopBar
          location={undefined}
        />,
        <styled.div>
          <styled.div>
            Who's Watching? 
          </styled.div>
          <styled.div>
            <Profile
              img="rick.jpg"
              userName="Rick"
            />
            <Profile
              img="morty.png"
              userName="Morty"
            />
            <Profile
              img="meeseeks.jpg"
              userName="Mr. Meeseeks"
            />
            <Profile
              userName="Add Profile"
            />
          </styled.div>
          <styled.span>
            Manage Profiles
          </styled.span>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "location": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              Who's Watching? 
            </styled.div>,
            <styled.div>
              <Profile
                img="rick.jpg"
                userName="Rick"
              />
              <Profile
                img="morty.png"
                userName="Morty"
              />
              <Profile
                img="meeseeks.jpg"
                userName="Mr. Meeseeks"
              />
              <Profile
                userName="Add Profile"
              />
            </styled.div>,
            <styled.span>
              Manage Profiles
            </styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Who's Watching? ",
            },
            "ref": null,
            "rendered": "Who's Watching? ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Profile
                  img="rick.jpg"
                  userName="Rick"
                />,
                <Profile
                  img="morty.png"
                  userName="Morty"
                />,
                <Profile
                  img="meeseeks.jpg"
                  userName="Mr. Meeseeks"
                />,
                <Profile
                  userName="Add Profile"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "img": "rick.jpg",
                  "userName": "Rick",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "img": "morty.png",
                  "userName": "Morty",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "img": "meeseeks.jpg",
                  "userName": "Mr. Meeseeks",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "userName": "Add Profile",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Manage Profiles",
            },
            "ref": null,
            "rendered": "Manage Profiles",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TopBar
            location={undefined}
          />,
          <styled.div>
            <styled.div>
              Who's Watching? 
            </styled.div>
            <styled.div>
              <Profile
                img="rick.jpg"
                userName="Rick"
              />
              <Profile
                img="morty.png"
                userName="Morty"
              />
              <Profile
                img="meeseeks.jpg"
                userName="Mr. Meeseeks"
              />
              <Profile
                userName="Add Profile"
              />
            </styled.div>
            <styled.span>
              Manage Profiles
            </styled.span>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "location": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                Who's Watching? 
              </styled.div>,
              <styled.div>
                <Profile
                  img="rick.jpg"
                  userName="Rick"
                />
                <Profile
                  img="morty.png"
                  userName="Morty"
                />
                <Profile
                  img="meeseeks.jpg"
                  userName="Mr. Meeseeks"
                />
                <Profile
                  userName="Add Profile"
                />
              </styled.div>,
              <styled.span>
                Manage Profiles
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Who's Watching? ",
              },
              "ref": null,
              "rendered": "Who's Watching? ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Profile
                    img="rick.jpg"
                    userName="Rick"
                  />,
                  <Profile
                    img="morty.png"
                    userName="Morty"
                  />,
                  <Profile
                    img="meeseeks.jpg"
                    userName="Mr. Meeseeks"
                  />,
                  <Profile
                    userName="Add Profile"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "img": "rick.jpg",
                    "userName": "Rick",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "img": "morty.png",
                    "userName": "Morty",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "img": "meeseeks.jpg",
                    "userName": "Mr. Meeseeks",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "userName": "Add Profile",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Manage Profiles",
              },
              "ref": null,
              "rendered": "Manage Profiles",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
